modules/standard/Types.chpl

```diff
@@ -95,36 +95,147 @@ proc isIntType(type t) param return
 proc isUintType(type t) param return
   (t == uint(8)) || (t == uint(16)) || (t == uint(32)) || (t == uint(64));
 
+/* Returns `true` if the type `t` is a `real` type, of any width. */
+proc isRealType(type t) param return
+  (t == real(32)) || (t == real(64));
+
+/* Returns `true` if the type `t` is an `imag` type, of any width. */
+proc isImagType(type t) param return
+  (t == imag(32)) || (t == imag(64));
+
+/* Returns `true` if the type `t` is a `complex` type, of any width. */
+proc isComplexType(type t) param return
+  (t == complex(64)) || (t == complex(128));
+
+/* Returns `true` if the type `t` is the `string` type. */
+proc isStringType(type t) param return t == string;
+
+/* Returns `true` if the type `t` is the `bytes` type. */
+proc isBytesType(type t) param return t == bytes;
+
 /* Returns `true` if the type `t` is an `enum` type. */
 proc isEnumType(type t) param {
   proc isEnumHelp(type t: enum) param return true;
   proc isEnumHelp(type t) param return false;
   return isEnumHelp(t);
 }
 
+/* Return true if ``t`` is a class type. Otherwise return false. */
+proc isClassType(type t) param return __primitive("is class type", t);
+
+/* Return true if ``t`` is a record type. Otherwise return false. */
+proc isRecordType(type t) param {
+  if __primitive("is record type", t) == false then
+    return false;
+
+  // some non-record types are implemented via records - exclude those
+  else if isDmapType(t)   ||
+          isDomainType(t) ||
+          isArrayType(t)  ||
+          isRangeType(t)  ||
+          isTupleType(t)  ||
+          isSyncType(t)   ||
+          isSingleType(t) ||
+          isStringType(t) ||
+          isAtomicType(t) then
+    return false;
+
+  else
+    return true;
+}
+
+/* Return true if ``t`` is a union type. Otherwise return false. */
+proc isUnionType(type t) param return __primitive("is union type", t);
+
+/*
+  Returns `true` if its argument is a tuple type.
+  The argument must be a type.
```

Comment:
Not sure important, but figured I should note this is out of keeping with the rest of the documentation